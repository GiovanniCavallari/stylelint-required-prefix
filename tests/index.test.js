const path = require('path');
const stylelint = require('stylelint');

function resolveFile(pathname, name) {
  return path.resolve(__dirname, pathname, name);
}

function extractErrorMessage(warnings) {
  return warnings.map(({ text }) => text);
}

describe('stylelint required prefix plugin', () => {
  it('should pass for valid css file', () => {
    return stylelint
      .lint({
        files: [resolveFile('fixtures', 'valid.css')],
        configFile: resolveFile('configs', 'stylelint.js'),
      })
      .then(({ results }) => {
        expect(results).toHaveLength(1);
        expect(results[0].warnings).toHaveLength(0);
        expect(extractErrorMessage(results[0].warnings)).toMatchSnapshot();
      });
  });

  it('should throw error for invalid css file', () => {
    return stylelint
      .lint({
        files: [resolveFile('fixtures', 'invalid.css')],
        configFile: resolveFile('configs', 'stylelint.js'),
      })
      .then(({ results }) => {
        expect(results).toHaveLength(1);
        expect(results[0].warnings).toHaveLength(3);
        expect(extractErrorMessage(results[0].warnings)).toMatchSnapshot();
      });
  });
});
