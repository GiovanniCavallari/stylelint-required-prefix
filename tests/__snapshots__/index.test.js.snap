// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylelint required prefix plugin should pass for valid css file 1`] = `[]`;

exports[`stylelint required prefix plugin should throw error for invalid css file 1`] = `
[
  "Expected prefix "gc-" on selector ".selector" (@gcavallari/stylelint-required-prefix)",
  "Expected prefix "gc-" on selector ".selector .primary:hover" (@gcavallari/stylelint-required-prefix)",
  "Expected prefix "gc-" on selector ".selector .secondary:hover" (@gcavallari/stylelint-required-prefix)",
]
`;
